static/js/admin.js

```diff
@@ -408,6 +432,51 @@ exports.setup_page = function () {
             }
         });
     });
+
+    $('.admin_emoji_table').on('click', '.delete', function (e) {
+        e.preventDefault();
+        e.stopPropagation();
+        var btn = $(this);
+
+        channel.del({
+            url: '/json/realm/emoji/' + encodeURIComponent(btn.attr('data-emoji-name')),
+            error: function (xhr, error_type) {
+                if (xhr.status.toString().charAt(0) === "4") {
+                    btn.closest("td").html(
+                        $("<p>").addClass("text-error").text($.parseJSON(xhr.responseText).msg)
+                    );
+                } else {
+                     btn.text("Failed!");
+                }
+            },
+            success: function () {
+                var row = btn.parents('tr');
+                row.remove();
+            }
+        });
+    });
+
+    $(".administration").on("submit", "form.admin-emoji-form", function (e) {
+        e.preventDefault();
+        e.stopPropagation();
+        var emoji_status = $('#admin-emoji-status');
+        var emoji_table = $('.admin_emoji_table');
+        var emoji = {};
+        $(this).serializeArray().map(function(x){emoji[x.name] = x.value;});
+
+        channel.put({
+            url: "/json/realm/emoji",
+            data: $(this).serialize(),
```

Comment:
I'm a little confused by how this code is using `$(this)`, can you explain it a bit?